interfaces/builtin/snap_refresh_inhibit_observe.go

```diff
@@ -0,0 +1,47 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+const snapRefreshInhibitObserveSummary = `allows read access to track snap refresh inhibition.`
+
+const snapRefreshInhibitObserveBaseDeclarationPlugs = `
+  snap-refresh-inhibit-observe:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const snapRefreshInhibitObserveBaseDeclarationSlots = `
+  snap-refresh-inhibit-observe:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+func init() {
+	registerIface(&commonInterface{
+		name:                 "snap-refresh-inhibit-observe",
```

Comment:
Not sure about the name, we're allowing access to more information than just refresh inhibition. You can now list all the changes and query their status with details, and lists all the snaps. How about `snap-refresh-observe` ?

Reply:
Makes sense, thank you! updated